pkg/projects/resource_project_membership.go

```diff
@@ -0,0 +1,214 @@
+package projects
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type Equatable interface {
+	Equals(other Equatable) bool
+}
+
+// Use by both project user and project group, as they shared identical data structure
+type Member struct {
+	Name  string   `json:"name"`
+	Roles []string `json:"roles"`
+}
+
+// Use by both project user and project group, as they shared identical data structure
+type Membership struct {
+	Members []Member
+}
+
+func (a Member) Equals(b Member) bool {
+	return a.Name == b.Name
+}
+
+func contains(as []Member, b Member) bool {
```

Comment:
You can actually define this on `Membership`

Reply:
Oh yeah. Let me try that.